test/specs/line.spec.js

```diff
@@ -1,3 +1,101 @@
 describe('Line annotation', function() {
   describe('auto', jasmine.fixtures('line'));
+
+  describe('inRange', function() {
+    const annotation1 = {
+      type: 'line',
+      scaleID: 'y',
+      value: 2,
+      borderWidth: 10
+    };
+    const annotation2 = {
+      type: 'line',
+      scaleID: 'x',
+      value: 8,
+      borderWidth: 5
+    };
+    const annotation3 = {
+      type: 'line',
+      scaleID: 'x',
+      value: 8,
+      borderWidth: 1
+    };
+    const annotation4 = {
+      type: 'line',
+      scaleID: 'x',
+      value: 8,
+      borderWidth: 0.5
+    };
+
+    const chart = window.scatterChart(10, 10, {annotation1, annotation2, annotation3, annotation4});
+    const elems = window.getAnnotationElements(chart);
+
+    elems.forEach(function(element) {
+      const center = element.getCenterPoint();
+      it('should return true inside element', function() {
+        const halfBorder = element.options.borderWidth / 2;
+        for (const x of [center.x - halfBorder, center.x, center.x + halfBorder]) {
+          for (const y of [center.y - halfBorder, center.y, center.y + halfBorder]) {
+            expect(element.inRange(x, y)).withContext(`scaleID: ${element.options.scaleID}, value: ${element.options.value}, center: {x: ${center.x.toFixed(1)}, y: ${center.y.toFixed(1)}}, {x: ${x.toFixed(1)}, y: ${y.toFixed(1)}}`).toEqual(true);
+          }
+        }
+      });
+
+      it('should return false outside element', function() {
+        const halfBorder = element.options.borderWidth / 2;
+        for (const x of [center.x - halfBorder - 1, center.x + halfBorder + 1]) {
+          for (const y of [center.y - halfBorder - 1, center.y + halfBorder + 1]) {
+            expect(element.inRange(x, y)).withContext(`scaleID: ${element.options.scaleID}, value: ${element.options.value}, center: {x: ${center.x.toFixed(1)}, y: ${center.y.toFixed(1)}}, {x: ${x.toFixed(1)}, y: ${y.toFixed(1)}}`).toEqual(false);
+          }
+        }
+      });
+    });
+  });
+
+  describe('inRange with rotation', function() {
+    const rotated = window.helpers.rotated;
+
+    for (const rotation of [0, 45, 90, 135, 180, 225, 270, 315]) {
+      const annotation = {
+        type: 'line',
+        scaleID: 'y',
+        value: 5,
+        label: {
+          enabled: true,
+          content: 'Label of the element',
+          rotation
+        }
+      };
+      const chart = window.scatterChart(10, 10, {annotation});
+      const element = window.getAnnotationElements(chart)[0];
+      it('should return true inside label of element', function() {
+        for (const borderWidth of [0, 10]) {
+          const halfBorder = borderWidth / 2;
+          element.options.label.borderWidth = borderWidth;
+          const rad = rotation / 180 * Math.PI;
+          for (const ax of [element.labelX - element.labelWidth / 2 - halfBorder + 1, element.labelX + element.labelWidth / 2 + halfBorder - 1]) {
```

Comment:
Similar issue (+-1) with the label `inRange`?

Reply:
it depends on the decimal places.

DEBUG: 

Object{x: 186.9892578125, y: 237.79999999999998}, <-- rotated event
Object{labelX: 256, labelY: 256, w2: 69.0107421875, h2: 18.2} <-- label

For instance, with 45 degrees,  the check that fail is `y >= labelY - h2` because

y = 237.79999999999998 and (labelY - h2) = 237.8

It needs to add a margin, adding 0.00001 or something like that.

Reply:
I use the same default espilo value of intersect: const epsilon = 0.001;

Ok for you?

Reply:
have a look if ok for you, using a constant 0.001 as default epsilon in intersect and isOnLabel methods